lib/scanner/walk_test.go

```diff
@@ -366,6 +366,43 @@ func TestWalkSymlinkWindows(t *testing.T) {
 	}
 }
 
+func TestWalkRootSymlink(t *testing.T) {
+
+	// Create a folder with a symlink to it
+
+	link := "link"
+
+	os.Remove(link)
+	defer os.Remove(link)
+
+	if err := osutil.DebugSymlinkForTestsOnly("testdata/dir1", link); err != nil && runtime.GOOS == "windows" {
+		// Probably we require permissions we don't have.
+		t.Skip("Need admin permissions or developer mode to run symlink test on Windows: " + err.Error())
+	}
+
+	// Scan it
+
+	fchan, err := Walk(context.TODO(), Config{
+		Filesystem: fs.NewFilesystem(fs.FilesystemTypeBasic, link),
+		BlockSize:  128 * 1024,
+	})
```

Comment:
The tests fail because you are missing `Hashers`. But you can use `walkDir` anyway, which takes care of such things and you don't need to loop over fchan later on:
```
files, err := walkDir(fs.NewFilesystem(fs.FilesystemTypeBasic, link), ".")
```